packages/salesforcedx-vscode-replay-debugger/src/states/variableAssignmentState.ts

```diff
@@ -29,15 +29,88 @@ export class VariableAssignmentState implements DebugLogState {
       const varName =
         nameSplit.length > 0 ? nameSplit[nameSplit.length - 1] : name;
       const value = this.fields[4];
+      let ref = '0';
+      if (this.fields.length === 6) {
+        ref = this.fields[5];
+      }
+      let container: ApexVariableContainer | undefined;
       if (logContext.getStaticVariablesClassMap().has(className)) {
         const statics = logContext.getStaticVariablesClassMap().get(className)!;
-        statics.get(name)!.value = value;
+        container = statics.get(name)! as ApexVariableContainer;
       } else if (frameInfo.locals.has(varName)) {
-        const frameLocals = frameInfo.locals.get(varName) as ApexVariable;
-        frameLocals.value = value;
+        // if name does not contain '.' (i.e. this.attr or a.Name), it should be in locals and we can update the value
+        container = frameInfo.locals.get(varName) as ApexVariableContainer;
+      }
+      if (container) {
+        container.value = value;
       }
-    }
 
+      // assigning to top level variable in locals
+      if (
+        ref !== '0' &&
+        container &&
+        !container.type.startsWith('Map<') &&
+        !container.type.startsWith('List<') &&
+        !container.type.startsWith('Set<')
+      ) {
+        if (!logContext.getRefsMap().has(ref)) {
+          logContext.getRefsMap().set(ref, container);
+        } else {
+          // if the ref already exists then that means we are assigning another pointer to a ref so we need to merge the info from the ref into the variable container
+          const refContainer = logContext.getRefsMap().get(ref)!;
+          container.variables = refContainer.variables;
+          container.variablesRef = refContainer.variablesRef;
+          container.name = varName;
+        }
+        if (value.indexOf('{') === 0 && value !== '{}') {
+          container.value = '';
+          container.variablesRef = logContext
+            .getVariableHandler()
+            .create(container);
+          this.parseAndPopulate(value, container);
+        }
+        // assigning to variable's fields, currently only going to work for a variable in local
+      } else if (name.indexOf('.') !== -1 && ref !== '0') {
```

Comment:
If we are assigning to an attribute of the variable then we get the container the ref is pointing to and add that attribute to the container's variables